File: src/chapters/introduction/content/code/horner.py
def horner(coeffs, x):
    """Evaluate a polynomial given coefficients [a_n, ..., a_0] (high -> low)."""
    y = coeffs[0]
    for a in coeffs[1:]:
        y = y * x + a
    return y


if __name__ == "__main__":
    # p(x) = 5x^4 - 8x^3 + 2x^2 + 4x - 10
    print(horner([5, -8, 2, 4, -10], 2))   # 22

File: src/pages/practice/guides/matlab/horner.m
function y = horner(a, x)
% HORNER  Evaluate a polynomial with coefficients a = [a_n ... a_0] (high->low).
    y = a(1);
    for i = 2:numel(a)
        y = y*x + a(i);
    end
end

% --- Demo ---  p(x) = 5x^4 - 8x^3 + 2x^2 + 4x - 10
disp(horner([5 -8 2 4 -10], 2));   % 22
